### PR TITLE
Prevents 'ipfs name publish' when /ipns is mounted.

### PR DESCRIPTION
**Pull request by noffle**
Addresses https://github.com/ipfs/go-ipfs/issues/1186.

### CODE REVIEW

File: test/sharness/t0031-mount-publish.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+test_expect_success "bootstrap to iptb peers" '
+  ipfs bootstrap add '$ADDR1' &&
+  ipfs bootstrap add '$ADDR2'
+'
+
+# launch the daemon
+test_launch_ipfs_daemon
+
+# wait for peer bootstrapping
+# TODO(noffle): this is very fragile -- how can we wait for this to happen for sure?
```
Comment (whyrusleeping) [inline test/sharness/t0031-mount-publish.sh]: check the other iptb tests to see how i wait for bootstrapping

Comment (whyrusleeping) [inline test/sharness/t0031-mount-publish.sh]: mostly, the functions in test/sharness/lib/iptb-lib.sh

**Comment by noffle** [comment]
Nice! `iptb` made this test way more solid. :+1:

File: test/sharness/t0031-mount-publish.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	skip_all='skipping mount tests, fuse not available'
+
+	test_done
+fi
+
+test_init_ipfs
+
+# start iptb + wait for peering
+NUM_NODES=3
+iptb init -n $NUM_NODES -f --bootstrap=none --port=0
```
Comment (whyrusleeping) [inline test/sharness/t0031-mount-publish.sh]: throw this in a test_expect_success

**Comment by whyrusleeping** [comment]
one small comment and LGTM

**Comment by noffle** [comment]
Looks like CI likes me. Ready for merge?

**Comment by whyrusleeping** [comment]
LGTM, wanna rebase and i'll merge?

**Comment by noffle** [comment]
Done.

**Comment by whyrusleeping** [comment]
Thanks!

**Pull request merged**